### PR TITLE
Make the rug keep trying if quantum does not respond

### PR DESCRIPTION
**Pull request by dhellmann**
Creates a new function the rug can use to wrap calls to
quantum in case the server isn't there or doesn't respond.
When that happens, the call is repeated until it succeeds.

bug: DHC-1052

Change-Id: I92b225f3687ac3a68d87934065f43bfc105867ab
Signed-off-by: Doug Hellmann doug.hellmann@dreamhost.com

### CODE REVIEW

**Comment by dhellmann** [comment]
I'll change that function so it only ignores some types of exceptions.

**Comment by dhellmann** [comment]
Updated to only ignore QuantumClientException when calling ensure_local_service_port().

**Comment by rodis** [comment]
LGTM

File: akanda/rug/manager.py [hunk 1]
```diff
@@ -143,10 +166,14 @@ def router_deleted(self, context, router_id=None):
     def sync_state(self):
         """Load state from database and update routers that have changed."""
         # pull all known routers
+        quantum_routers = wait_for_callable(
+            self.quantum.get_routers,
+            error_msg='Could not fetch routers from quantum',
```
Comment (markmcclain) [inline akanda/rug/manager.py:173]: Don't we need to add ignorable exceptions here?

Comment (dhellmann) [inline akanda/rug/manager.py:173]: The intent was to only use that for code in the startup path, and it wasn't obvious that sync_state() is called during startup.

Comment (dhellmann) [inline akanda/rug/manager.py:173]: Updated.

**Pull request merged**